src/ToricVarieties/ToricDivisors/attributes.jl

```diff
@@ -66,6 +67,19 @@ julia> coefficients(D)
 ```
 """
 function coefficients(td::ToricDivisor)
-    return Vector{Int}(Polymake.common.primitive(pm_tdivisor(td).COEFFICIENTS))
+    return get_attribute!(td, :coefficients) do
+        return Vector{Int}(Polymake.common.primitive(pm_tdivisor(td).COEFFICIENTS))
+    end
 end
 export coefficients
+
+
+@doc Markdown.doc"""
+    toricvariety(td::ToricDivisor)
+
+Return the toric variety of a torus-invariant Weil divisor.
+"""
+function toricvariety(td::ToricDivisor)
+    return get_attribute(td, :toricvariety)
```

Comment:
This will return `nothing` if `toricvariety` is not set. Is this intentional? Or can it never happen? But if it can never happen, then why is `toricvariety` even an attribute instead of a regular struct field?

If it really should stay an attribute, at least add a type assertion?
```suggestion
    return get_attribute(td, :toricvariety)::AbstractNormalToricVariety
```

Reply:
> Looks good to me.
> 
> From my POV we could merge now; the things I mention are truly minor. I'll do it tomorrow or the day after if I hear no complaints

Thank you @fingolfin .

Reply:
You are right - let me turn this into a struct field.